src/client/mod.rs

```diff
@@ -99,66 +145,144 @@ impl<Resolver: SrvResolver, Policy: policy::Policy + Default> SrvClient<Resolver
     }
 
     /// Gets a client's cached items, refreshing the existing cache if it is invalid.
-    pub async fn get_valid_cache_items(
+    async fn get_valid_cache(
         &self,
-    ) -> Result<impl Deref<Target = [Policy::CacheItem]>, SrvError<Resolver::Error>> {
-        let cache = match self.cache.load() {
-            cache if cache.valid() => cache,
-            _ => Guard::from_inner(self.refresh_cache().await?),
-        };
-        Ok(Cache::items(cache))
+    ) -> Result<Arc<Cache<Policy::CacheItem>>, SrvError<Resolver::Error>> {
+        match self.cache.load_full() {
+            cache if cache.valid() => Ok(cache),
+            _ => self.refresh_cache().await,
+        }
     }
 
-    /// Determines an ordering of URIs to select based on the current [`Policy`].
+    /// Performs an operation on a client's SRV targets, producing a stream of
+    /// results.
```

Comment:
This should be a bit more explicit about executing on *all* of the targets.

Reply:
```rust
/// Performs an operation on all of a client's SRV targets, producing a
/// stream of results (one for each target). If the serial execution mode is
/// specified, the operation will be performed on each target in the order
/// determined by the current [`Policy`], and the results will be returned
/// in the same order. If the concurrent execution mode is specified, the
/// operation will be performed on all targets concurrently, and results
/// will be returned in the order they become available.
```
Better?